shapely/ops.py

```diff
@@ -718,3 +723,50 @@ def orient(geom, sign=1.0):
 
     """
     return shapely.orient_polygons(geom, exterior_cw=sign < 0)
+
+
+def get_segments(geometry):
```

Comment:
Not completely sure about this, but looking at the tests being in the `legacy` folder, I think the `ops.py` file is there for historic reasons.  More recently we are trying to put the algorithms into categories. In this case I think the function should best be placed in the same file as `get_coordinates`. The tests may be moved as well.

Reply:
In shapely, many functions are thin wrappers around C-implemented ones. Take for instance the `get_coordinates`: its implementation is just a  direct call into `lib.get_coordinates`:

```
lib.get_coordinates(
        np.asarray(geometry, dtype=np.object_), include_z, include_m, return_index
    )
```

What this wrapper adds is 1) type coercion and 2) docstrings. You don't need those, as long as you are doing the type coercion correctly if necessary.

So I suggest not depending on `creation.py` and `coordinates.py`, but directly calling into `lib.get_coordinates` and `lib.linestrings`. Most modules already depend on `lib` so that will be fine. Then you'll have free choice of where to put the function. 

My first choice would be `_geometry.py` next to `get_parts`.